packages/template-retail-react-app/app/components/product-tile/index.test.js

```diff
@@ -129,3 +129,125 @@ test('renders strike through price with standard product', () => {
     expect(currentPriceTag).toHaveLength(1)
     expect(strikethroughPriceTag).toHaveLength(1)
 })
+
+test('renders badges corresponding to the default custom properties', () => {
+    const {getByText, container} = renderWithProviders(
+        <ProductTile product={mockStandardProductHit} />
+    )
+    expect(getByText(/Laptop Briefcase with wheels \(37L\)/i)).toBeInTheDocument()
+    const badges = container.querySelectorAll('span')
+    expect(badges).toHaveLength(3)
+    expect(within(badges[0]).getByText(/New/i)).toBeDefined()
+    expect(within(badges[1]).getByText(/Sale/i)).toBeDefined()
```

Comment:
I think this can be confusing that we're expecting to see 3 spans, while only checking the first 2. I'm guessing that not all spans are the actual badges?

If that's the case, we can actually use a test id to identify the Badge component... something like this:
```js
// In the component file
// <Badge data-testid="badge" />

// In the test file
const {getAllByTestId} = renderWithProviders(...)
const badges = getAllByTestId('badge')
expect(badges).toHaveLength(2)
```